pkg/operator/baremetal_pod.go

```diff
@@ -5,10 +5,23 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/pointer"
+
+	"encoding/json"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"math/rand"
+	"time"
+)
+
+const (
+	baremetalConfigmap = "metal3-config"
+	sharedVolume       = "metal3-shared"
+	sharedSecret       = "metal3-password"
+	secretName         = "mariadb-password"
+	secretKey          = "password"
 )
 
-var baremetalConfigmap = "metal3-config"
-var sharedVolume = "metal3-shared"
+var mariadbPassword = generateRandomPassword()
```

Comment:
The timing for generating this value is an interesting problem. Doing it here means that every time the machine-api-operator restarts, we will end up with a new value. Is that what we want? Will that cause an update to the Secret, or Deployment? Can the baremetal code handle an update to the Secret?

Can we create a new password only if the Secret doesn't exist or doesn't have a valid password key? That would mean checking for an existing Secret before creating a new one and not updating it if it does exist, which complicates the logic but I think will make this more robust.

Reply:
Yeah, seems like we could just leave the name as something predictable e.g mariadb-password, and only create it when it doesn't exist?

Reply:
Updated my diffs to create the password only when it doesn't already exist in the namespace.